ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLFlowAgentRunner.java

```diff
@@ -145,6 +148,51 @@ public void run(MLAgent mlAgent, Map<String, String> params, ActionListener<Obje
         }
     }
 
+    private void updateMemory(Map<String, Object> additionalInfo, String memoryType, String memoryId) {
+        ConversationIndexMemory.Factory conversationIndexMemoryFactory = (ConversationIndexMemory.Factory) memoryFactoryMap.get(memoryType);
+        conversationIndexMemoryFactory.create(memoryId, ActionListener.<ConversationIndexMemory>wrap(memory -> {
+            updateInteraction(additionalInfo, memoryId, memory);
+        }, e -> { log.error("Failed create memory from id: " + memoryId, e); }));
+    }
+
+    private void updateInteraction(Map<String, Object> additionalInfo, String memoryId, ConversationIndexMemory memory) {
+        memory
+            .getMemoryManager()
+            .getFinalInteractions(memoryId, PREVIOUS_INTERACTION, ActionListener.<List<Interaction>>wrap(interactions -> {
+                if (interactions.size() == 0) {
+                    throw new IllegalStateException("No existing interactions to update");
+                }
```

Comment:
Good question:
1. Will update to create an empty interaction in case there are no existing interaction. 
2. As per my understanding `getFinalInteraction` will return previous interaction with final answer. @Zhangxunmt please confirm if `getFinalInteractions` can result in race condition.

Reply:
getFinalInteractions returns all interactions that are not traces. The final interaction itself is always created when an interaction starts, with an empty response and it waits until all the traces are complete to obtain the final answer. So it's possible that new interaction is created before the previous one receives the final response.